hs/app/reachpc/ReachPC/Config.hs

```diff
@@ -0,0 +1,170 @@
+  globalRchToml <- readOrCreateGlobalReachToml
+  envVars <- getEnvironment
+  let envVar = flip lookup envVars
+  
+  -- The pattern here is that for every part of the project config, we check (and prioritize)
+  -- env var, command line flag, project reach.toml, global reach.toml.
+  
+  -- Whether to use reachd-cloud or reachd-local
+  -- REACH_CLOUD_OR_LOCAL / --cloud or --local / cloud-or-local = "cloud" or "local"
+  let env_cloudOrLocal = readCloudOrLocal <$> envVar "REACH_CLOUD_OR_LOCAL"
+  let cli_cloudOrLocal = boolToCloudOrLocal <$> Cli.cli_cloudOrLocal cliOpts
+  let prj_cloudOrLocal = rtml_cloudOrLocal projectRchToml
+  let glb_cloudOrLocal = rtml_cloudOrLocal globalRchToml
+  let cfg_cloudOrLocal = unwrapConfigOption "Unspecified whether to use Reach Cloud or Reach Local." $
+                         env_cloudOrLocal <|> cli_cloudOrLocal <|> prj_cloudOrLocal <|> glb_cloudOrLocal
+
+  -- What connector to use (algo/cfx/eth/...)
+  -- REACH_CONNECTOR / --connector=... / connector = "algo" or "eth" or ...
+  let env_connector = readConnector <$> envVar "REACH_CONNECTOR"
+  let cli_connector = readConnector <$> Cli.cli_connector cliOpts
+  let prj_connector = rtml_connector projectRchToml
+  let glb_connector = rtml_connector globalRchToml
+  let cfg_connector = unwrapConfigOption "Unspecified what connector to use." $
+                      env_connector <|> cli_connector <|> prj_connector <|> glb_connector
+
+  -- What env vars to forward to the remote (sources are combined, not prioritized)
+  -- REACH_FORWARD_ENV_VARS / --env / forward-env-vars = [...]
+  let env_forwardEnvVars = maybe [] (splitOn ",") $ envVar "REACH_FORWARD_ENV_VARS"
+  let cli_forwardEnvVars = fromMaybe [] $ Cli.cli_forwardedEnvVars cliOpts
+  let prj_forwardEnvVars = fromMaybe [] $ rtml_forwardEnvVars projectRchToml
+  let gbl_forwardEnvVars = fromMaybe [] $ rtml_forwardEnvVars globalRchToml
+  let cfg_forwardEnvVars = env_forwardEnvVars <> cli_forwardEnvVars <> prj_forwardEnvVars <> gbl_forwardEnvVars
+  
+  -- Remote project identifer
+  -- REACH_PROJECT / --project=.. / project = ".."
+  let env_project = envVar "REACH_PROJECT"
+  let cli_project = Cli.cli_project cliOpts
+  let prj_project = rtml_project projectRchToml
+  let glb_project = rtml_project globalRchToml
+  let cfg_project = unwrapConfigOption "Unspecified remote project identifier." $
+                    env_project <|> cli_project <|> prj_project <|> glb_project
+
+  -- Remote organization
+  -- REACH_ORGANIZATION / --organization=... / organization = ".."
+  let env_organization = envVar "REACH_ORGANIZATION"
+  let cli_organization = Cli.cli_organization cliOpts
+  let prj_organization = rtml_organization projectRchToml
+  let glb_organization = rtml_organization globalRchToml
+  let cfg_organization = unwrapConfigOption "Unspecified remote organization." $
+                         env_organization <|> cli_organization <|> prj_organization <|> glb_organization
```

Comment:
Abstraction... but also remember my comment about the org coming from the project